### PR TITLE
run CI on julia 1.11

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
Updates everything to use Julia v1.11

Closes #1001 

## Content
- [x] update all CI workflows to Julia v1.11
- [x] update compat entries to use 1.11-compatible versions
- [x] update dependencies in existing Manifests
- [x] generate new Manifests for Julia v1.11
- [x] update package README and NEWS to reflect these changes

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -29,3 +29,13 @@ to our output diagnostics using the ClimaDiagnostics interface.
...
 functionality is replaced by the generalized `make_plots` function.
+
+#### Remove PostProcessor module - PR [#1022](https://github.com/CliMA/ClimaCoupler.jl/pull/1022)
+After switching to use ClimaDiagnostics.jl and ClimaAnalysis.jl
+for our diagnostics, the PostProcessor module is not needed.
+This PR follows directly from the Diagnostics module removal.
+
+### Maintenance
+- Update to JuliaFormatter v2. PR [#1024](https://github.com/CliMA/ClimaCoupler.jl/pull/1024)
+- Update CI to use Julia v1.11. Introduce Manifest files for
```
Comment (akshaysridhar) [inline NEWS.md:40]: Thanks @juliasloan25 !

File: NEWS.md [hunk 2]
```diff
@@ -29,3 +29,13 @@ to our output diagnostics using the ClimaDiagnostics interface.
...
 This PR also removes the AMIP paperplots function, but this
 functionality is replaced by the generalized `make_plots` function.
+
+#### Remove PostProcessor module - PR [#1022](https://github.com/CliMA/ClimaCoupler.jl/pull/1022)
+After switching to use ClimaDiagnostics.jl and ClimaAnalysis.jl
+for our diagnostics, the PostProcessor module is not needed.
+This PR follows directly from the Diagnostics module removal.
+
+### Maintenance
+- Update to JuliaFormatter v2. PR [#1024](https://github.com/CliMA/ClimaCoupler.jl/pull/1024)
```
Comment (Sbozzolo) [inline NEWS.md:39]: The release for JuliaFormatter v2 was yanked because it broke some existing codes

**Comment by Sbozzolo** [comment]
It looks like that there's still something not updated because the last job on buildkite failed with a `StyledString` error

**Review by Sbozzolo — APPROVED** [review]

**Comment by juliasloan25** [comment]
> It looks like that there's still something not updated because the last job on buildkite failed with a `StyledString` error

This was due to an error on Central that IMSS is working on fixing. Nat has implemented a workaround for now

**Pull request merged**